src/Arcus.Messaging.Health/Tcp/TcpHealthListener.cs

```diff
@@ -127,8 +119,7 @@ private async Task AcceptConnectionAsync()
                     string clientId = client.Client?.RemoteEndPoint?.ToString() ?? String.Empty;
                     _logger.LogInformation("Return '{Status}' health report to client {ClientId}", report.Status, clientId);
 
-                    string serialized = JsonConvert.SerializeObject(report, _serializerSettings);
-                    byte[] response = Encoding.UTF8.GetBytes(serialized);
+                    byte[] response = _tcpListenerOptions.ReportSerializer(report);
```

Comment:
Do we use this approach, or should people assign a different serializer to our options?

Ie, `TcpHealthListenerOptions.Serializer` could be our default unless they change it to another one

Reply:
The default `ReportSerializer` is set as default, and will be used if the user doesn't change it with `.Configure(options => options.ReportSerializer = ...)` .

Reply:
I know, but now it's a function - Doesn't it feel more natural if you just give it a deserializer rather than implementation?

Reply:
:D FP vs OOP.
`ISerializer` is a .NET interface I think? Would you rather have that we use that one?

Reply:
It might be just me, what do you think @fgheysels?

Reply:
Hmm, the message pump has also some functions instead of objects for settings, we should consider use the same approach?

Reply:
That's different imo - The message pump allows you to configure the message pump by mutating the default/existing configuration by just changing the object.

Here we are doing a whole lot of processing by passing a lambda, this is not really OOP as you've mentioned

Reply:
So introducing two interfaces and two default implementations?

Reply:
Why two?

Reply:
To retrieve the TCP health port and to set the serialization logic to parse health reports.

Reply:
TCP health port should just be a configuration key name in the extension such as:
```csharp
services.AddTcpHealthProbes("ARCUS_HEALTH_PORT");
```

This should be more than enough for now, if people need more control we can still open it up later on!

Commented this suggestion somewhere this morning but can't seem to find it anymore :/

Reply:
For the other one an interface would be sufficient for me. If the `CustomSerializer` or so is not specified we can use the default implementation without the additional wrapper.

Don't think we should create an default implementation which just calls our dependent library imo but I'm open to anything

Reply:
Added a suggestion.